### ISSUE TITLE
Persist workspace assignments without modifying the config

### ISSUE THREAD

**Issue by vfaberd** [nikitabobko/AeroSpace#57]
Hi, thank you for this amazing tool! I love how lightweight and unobtrusive it is.

**Question:**
Is there a way to persist assignments to workspaces other than adding each individual app to the config? I have a lot of apps that I do not use often and that are specific to only one machine. Consequently, I'd rather avoid cluttering my config with these apps. However, I still want them to remain in the workspaces where I put them after reboots or application restarts.

**Comment by nikitabobko** [comment]
Reference: https://i3wm.org/docs/layout-saving.html

**Comment by gogamid** [comment]
Of course, having a layout file as in i3 would be the dream. Until it’s implemented, I’ve found a handy workaround. I use a bash script, triggered via Alfred, to set up the simple layouts that I frequently use. For instance, I have setups for different scenarios like tutorials, thesis writing, or work workspaces. Just thought I’d share this tip while we wait for the layout file feature. Hope this helps! 
```bash
#!/bin/bash
view=$1
if [ "$view" == "tutorial" ]; then
	ws=1
	open -a Firefox
	aerospace move-node-to-workspace $ws

	open -a IINA
	sleep 1
	aerospace layout tiling
	aerospace move-node-to-workspace $ws

	open -a WezTerm
	aerospace move-node-to-workspace $ws

	aerospace workspace $ws
	aerospace layout tiling
	for i in {1..3}; do
		aerospace move left
	done
fi

if [ "$view" == "thesis" ]; then
	ws=2
	open -a sioyek
	sleep 1
	aerospace move-node-to-workspace $ws

	open -a WezTerm
	aerospace move-node-to-workspace $ws

	open -a Firefox
	aerospace move-node-to-workspace $ws

	aerospace workspace $ws
	aerospace layout tiling
	for i in {1..3}; do
		aerospace move left
	done
fi

function open_move_fullscreen {
	open -a $1
	aerospace move-node-to-workspace $2
	aerospace fullscreen
}

if [ "$view" == "default" ]; then
	aerospace close-all-windows-but-current
	open_move_fullscreen "ChatAll" A
	open_move_fullscreen Firefox B
	open_move_fullscreen WezTerm T
	aerospace workspace B
fi

if [ "$view" == "work" ]; then
	ws=C
	open -a "Microsoft Teams"
	sleep 1
	aerospace move-node-to-workspace $ws

	open -a "Microsoft Outlook"
	sleep 1
	aerospace move-node-to-workspace $ws

	aerospace workspace $ws
	aerospace layout tiling
fi
```

**Comment by januz** [comment]
Layout files would be fantastic! This is the feature I'm waiting most eagerly for (and others in this and also [this](https://github.com/nikitabobko/AeroSpace/issues/187) issue).

@nikitabobko Do you still have this feature on the roadmap? Thanks!

**Comment by bdlow** [comment]
I've created this small script to save and restore window -> workspace config; I run `aerospace-save_restore.sh save` before a aerospace "cold restart" (e.g. aerospace version update) and `aerospace-save_restore.sh restore` after. It's only useful when window IDs persist, so won't be useful to recover from OS updates/reboots.

https://gist.github.com/bdlow/59a35c8adeed8294929f1f6f8b0e1de7

**Comment by paragbaxi** [comment]
> I've created this small script to save and restore window -> workspace config; I run `aerospace-save_restore.sh save` before a aerospace "cold restart" (e.g. aerospace version update) and `aerospace-save_restore.sh restore` after. It's only useful when window IDs persist, so won't be useful to recover from OS updates/reboots.
> 
> https://gist.github.com/bdlow/59a35c8adeed8294929f1f6f8b0e1de7

gist is deleted?

**Comment by af3556** [comment]
> gist is deleted?

Fixed: https://gist.github.com/af3556/59a35c8adeed8294929f1f6f8b0e1de7

Unfortunately I've found it to be fairly limited utility: one of the other common annoyances for me is updating the browser and losing the placement of all windows; unfortunately just like a OS reboot the app restarting will renew all window IDs, thus voiding any possibility for "save and restore".

An effective "save and restore" functionality needs aerospace to be able to support a "by window title" for `aerospace move-node-to-workspace`.
